### PR TITLE
[BEAM-2566] Decouple SDK harness from Dataflow runner by elevating experiments and SDK harness configuration to java-sdk-core.

### PR DESCRIPTION
**Pull request by lukecwik**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
I'm entertaining name changes on the options interfaces or any suggestions on how to make it easier to migrate from DataflowWorkerLoggingOptions to SdkHarnessOptions for logging configuration.

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @tgroh 
CC: @kennknowles

File: runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/ArtifactServiceStagerTest.java [hunk 1]
```diff
@@ -45,10 +45,13 @@
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
 
 /**
  * Tests for {@link ArtifactServiceStager}.
  */
+@RunWith(JUnit4.class)
```
Comment (kennknowles) [inline runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/ArtifactServiceStagerTest.java]: This change seems unrelated. Can you break it into a separate commit?

Comment (lukecwik) [reply runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/ArtifactServiceStagerTest.java]: I can, just going to wait for the tests to pass on this giving me the green signal before.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.beam.sdk.annotations.Experimental;
+
+/**
+ * Options that are used to control logging configuration on the SDK harness.
+ */
+@Experimental
+@Description("Options that are used to control logging configuration on the SDK harness.")
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java]: This looks overspecified or the class name is too general.

Comment (lukecwik) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java]: Your right, will fix the description. Was copy paste from DataflowWorkerLoggingOptions.

**Comment by tgroh** [comment]
retest this please

**Comment by lukecwik** [comment]
retest this please

**Comment by lukecwik** [comment]
retest this please

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Everything succeeded except for the coveralls report at the very end, see:
https://builds.apache.org/job/beam_PreCommit_Java_MavenInstall/15337/console
